From patchwork Wed Oct 17 14:46:57 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Uros Bizjak <ubizjak@gmail.com>
X-Patchwork-Id: 10645731
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E458B157A
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 14:47:16 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D2228256E6
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 14:47:16 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id C249725EA6; Wed, 17 Oct 2018 14:47:16 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 63154256E6
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Oct 2018 14:47:16 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727405AbeJQWnQ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 17 Oct 2018 18:43:16 -0400
Received: from mail-wr1-f68.google.com ([209.85.221.68]:38196 "EHLO
        mail-wr1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727032AbeJQWnQ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 17 Oct 2018 18:43:16 -0400
Received: by mail-wr1-f68.google.com with SMTP id a13-v6so30033637wrt.5
        for <kvm@vger.kernel.org>; Wed, 17 Oct 2018 07:47:14 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=TY9PlLpB535qA+vqJoeNupa0i3x89iP4WaX7IQO3fSI=;
        b=O21DLj4pXaF1qeqTuea7splXjSeE1k0Ii2vwmOgdQckplFfF3Lk7OeH1KWRweOcRRc
         QuT2N3xKH0ZMngSxhS8RfBqhyFH+nlvgidPJ3bNlDtnAY0Orz6dkeZ8RY0pAFM1SSNNF
         LbcNg0saDaAwQWyd3fJvGDnNLnwGkH4PVuXLnAFtLJvUymIFk36x0yUFPpVEV+xP5grn
         4lFliLfzc1ZwOGvpS2J9cgzg4Sd82UcDMMAwsvDLw78FCgOogmJj+11/s98/38JJthxg
         t8yG/+4jRlRBYnkxChwfXRLqr2DKoJ0aan+yU2Lz12EYbNrcPY++gpU72nqKQtyJcai6
         o4Eg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=TY9PlLpB535qA+vqJoeNupa0i3x89iP4WaX7IQO3fSI=;
        b=kS4LOi3RBasPNdVoRrNwFAcq2FDcxLkXFYq0vEC9DPYixS2YMeS0+yfxUIz2UrcAPk
         o+0qa6q5EhVHU7M3s0KrArD9H0mnUzK+WgBGpQ6h9hv1eveMTFP0XWIX5y3aXAn8JUN/
         qF2+mepiHQ5NFg8URmWPlCrZ09Bb10KCa+cPIg17PlU+puRLVTcXiba8oGWwD+AYjoaJ
         aWVNdOTaWWubJdf7v9ThdrS9kqybHD6e9/KnY0VNxyqjPDMw4FYQXSpq7Pso5DduH2up
         RRT3nl2dLDjlNDCucu76883cavfv6IpY0Gaq/FaoBvh4gnBknW5TQY0HraPqhakh2jjM
         +Uxw==
X-Gm-Message-State: ABuFfohgvQs8yfxFEwwN0Hg5MEOmQU5jjwpnPBLbU/LBuxINcpmmuEGa
        EBZ4Qf9xD7hpqEjaz0Ok7AtVUQf/
X-Google-Smtp-Source: 
 ACcGV60q0Lb/WOi3cmnrjHYUkE747TQXHJ4TmcpjolzRy6gVBXLCyUJkk2MbG7Oq/Vu3zOnknSgChw==
X-Received: by 2002:adf:f384:: with SMTP id
 m4-v6mr2618014wro.111.1539787632957;
        Wed, 17 Oct 2018 07:47:12 -0700 (PDT)
Received: from localhost.localdomain.com (93-103-18-160.static.t-2.net.
 [93.103.18.160])
        by smtp.gmail.com with ESMTPSA id
 203-v6sm2635517wmn.34.2018.10.17.07.47.11
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Wed, 17 Oct 2018 07:47:12 -0700 (PDT)
From: Uros Bizjak <ubizjak@gmail.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Uros Bizjak <ubizjak@gmail.com>
Subject: [PATCH] KVM/x86: Use 32bit xor to clear registers in svm.c
Date: Wed, 17 Oct 2018 16:46:57 +0200
Message-Id: <20181017144657.3799-1-ubizjak@gmail.com>
X-Mailer: git-send-email 2.17.2
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

x86_64 zero-extends 32bit xor operation to a full 64bit register.

Also add a comment and remove unnecessary instruction suffix in vmx.c

Signed-off-by: Uros Bizjak <ubizjak@gmail.com>
---
 arch/x86/kvm/svm.c | 28 +++++++++++++---------------
 arch/x86/kvm/vmx.c |  6 +++++-
 2 files changed, 18 insertions(+), 16 deletions(-)

diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 1086bebaeea3..f416f5c7f2ae 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -5637,26 +5637,24 @@ static void svm_vcpu_run(struct kvm_vcpu *vcpu)
 		"mov %%r13, %c[r13](%[svm]) \n\t"
 		"mov %%r14, %c[r14](%[svm]) \n\t"
 		"mov %%r15, %c[r15](%[svm]) \n\t"
-#endif
 		/*
 		* Clear host registers marked as clobbered to prevent
 		* speculative use.
 		*/
-		"xor %%" _ASM_BX ", %%" _ASM_BX " \n\t"
-		"xor %%" _ASM_CX ", %%" _ASM_CX " \n\t"
-		"xor %%" _ASM_DX ", %%" _ASM_DX " \n\t"
-		"xor %%" _ASM_SI ", %%" _ASM_SI " \n\t"
-		"xor %%" _ASM_DI ", %%" _ASM_DI " \n\t"
-#ifdef CONFIG_X86_64
-		"xor %%r8, %%r8 \n\t"
-		"xor %%r9, %%r9 \n\t"
-		"xor %%r10, %%r10 \n\t"
-		"xor %%r11, %%r11 \n\t"
-		"xor %%r12, %%r12 \n\t"
-		"xor %%r13, %%r13 \n\t"
-		"xor %%r14, %%r14 \n\t"
-		"xor %%r15, %%r15 \n\t"
+		"xor %%r8d, %%r8d \n\t"
+		"xor %%r9d, %%r9d \n\t"
+		"xor %%r10d, %%r10d \n\t"
+		"xor %%r11d, %%r11d \n\t"
+		"xor %%r12d, %%r12d \n\t"
+		"xor %%r13d, %%r13d \n\t"
+		"xor %%r14d, %%r14d \n\t"
+		"xor %%r15d, %%r15d \n\t"
 #endif
+		"xor %%ebx, %%ebx \n\t"
+		"xor %%ecx, %%ecx \n\t"
+		"xor %%edx, %%edx \n\t"
+		"xor %%esi, %%esi \n\t"
+		"xor %%edi, %%edi \n\t"
 		"pop %%" _ASM_BP
 		:
 		: [svm]"a"(svm),
diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 91f4f5288150..3677609145a4 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -62,7 +62,7 @@
 
 #define __ex(x) __kvm_handle_fault_on_reboot(x)
 #define __ex_clear(x, reg) \
-	____kvm_handle_fault_on_reboot(x, "xorl " reg " , " reg)
+	____kvm_handle_fault_on_reboot(x, "xor " reg ", " reg)
 
 MODULE_AUTHOR("Qumranet");
 MODULE_LICENSE("GPL");
@@ -11285,6 +11285,10 @@ static void __noclone vmx_vcpu_run(struct kvm_vcpu *vcpu)
 		"mov %%r13, %c[r13](%0) \n\t"
 		"mov %%r14, %c[r14](%0) \n\t"
 		"mov %%r15, %c[r15](%0) \n\t"
+		/*
+		* Clear host registers marked as clobbered to prevent
+		* speculative use.
+		*/
 		"xor %%r8d,  %%r8d \n\t"
 		"xor %%r9d,  %%r9d \n\t"
 		"xor %%r10d, %%r10d \n\t"
